enterprise/api/v1/views/enterprise_customer_sso_configuration.py

```diff
@@ -59,3 +78,115 @@ def oauth_orchestration_complete(self, request, configuration_uuid, *args, **kwa
+        else:
+            return Response({'error': 'Must provide valid enterprise customer'}, status=HTTP_400_BAD_REQUEST)
+        try:
+            new_record = EnterpriseCustomerSsoConfiguration.objects.create(**request_data)
+        except TypeError as e:
+            LOGGER.error(f'Error creating SSO configuration record: {e}')
+            return Response({'error': f'Error creating SSO configuration record: {e}'}, status=HTTP_400_BAD_REQUEST)
+        return Response({'data': new_record.pk}, status=HTTP_201_CREATED)
+
+    @permission_required(
+        'enterprise.can_access_admin_dashboard',
+        fn=lambda request, *args, **kwargs: get_customer_from_request(request)
+    )
+    def update(self, request, *args, **kwargs):
+        configuration_uuid = kwargs.get('configuration_uuid')
+        sso_configuration_record = self.queryset.filter(pk=configuration_uuid)
+        if not sso_configuration_record:
+            return Response(status=HTTP_404_NOT_FOUND)
+        # Make sure the requesting user is part of the enterprise customer associated with the configuration record
+        # or are a staff member
+        config_customer = sso_configuration_record.first().enterprise_customer
+        if not check_user_part_of_customer(request.user, config_customer):
+            return Response(status=HTTP_404_NOT_FOUND)
+
+        # If the request includes a customer uuid, ensure the new customer is valid
+        if new_customer := request.data.dict().get('enterprise_customer'):
+            try:
+                enterprise_customer = EnterpriseCustomer.objects.get(uuid=new_customer)
+            except EnterpriseCustomer.DoesNotExist:
+                return Response({'error': 'Must provide valid enterprise customer'}, status=HTTP_400_BAD_REQUEST)
+            if not check_user_part_of_customer(request.user, enterprise_customer):
+                return Response({'error': 'Must provide valid enterprise customer'}, status=HTTP_400_BAD_REQUEST)
+
+        try:
+            sso_configuration_record.update(**request.data.dict())
+        except TypeError as e:
+            LOGGER.error(f'Error updating SSO configuration record: {e}')
+            return Response({'error': f'Error updating SSO configuration record: {e}'}, status=HTTP_400_BAD_REQUEST)
+        serializer = self.serializer_class(sso_configuration_record.first())
+        return Response(serializer.data, status=HTTP_200_OK)
+
+    @permission_required(
+        'enterprise.can_access_admin_dashboard',
+        fn=lambda request, *args, **kwargs: get_customer_from_request(request)
+    )
+    def destroy(self, request, *args, **kwargs):
+        configuration_uuid = kwargs.get('configuration_uuid')
+        sso_configuration_record = self.queryset.filter(pk=configuration_uuid)
+        if not sso_configuration_record:
+            return Response(status=HTTP_404_NOT_FOUND)
+        sso_configuration_record.update(is_removed=True)
```

Comment:
Do we need to check if the user is authorized to delete?

Reply:
In places above(like update) we use that same permission wrapper, but in addition check to make sure the person is part of the customer that owns the configuration being updated.  I would expect the same check if we're deleting a configuration.